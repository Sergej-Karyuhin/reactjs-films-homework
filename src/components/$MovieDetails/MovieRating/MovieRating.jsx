import React from 'react';
import PropTypes from 'prop-types';

import styles from './MovieRating.scss';

const MovieRating = ({ rating }) => {
  const STARS = new Array(Math.round(rating / 2)).fill().map(item => Math.round(Math.random() * 9999));
  return (
    <ul className={styles.stars}>
      {
        STARS.map( (star) =>
          <li className={styles.star} key={ star }></li>
        )
      }
      <div className={styles.rating}>{ rating }</div>
    </ul>
  );
}

MovieRating.propTypes = {
  rating: PropTypes.number.isRequired,
};

export default MovieRating;
