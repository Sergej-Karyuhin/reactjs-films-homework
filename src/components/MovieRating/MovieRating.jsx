import React from 'react';
import styles from './MovieRating.scss';

import PropTypes from 'prop-types';

const MovieRating = ({ movieRating }) => {
  const STARS = new Array(Math.round(movieRating)).fill().map(item => Math.round(Math.random() * 9999));
  return (
    <ul className={styles.stars}>
      {
        STARS.map( (star) =>
          <li className={styles.star} key={ star }></li>
        )
      }
      <div className={styles.rating}>{ movieRating }</div>
    </ul>
  );
}

MovieRating.propTypes = {
  movieRating: PropTypes.number,
};

export default MovieRating;
