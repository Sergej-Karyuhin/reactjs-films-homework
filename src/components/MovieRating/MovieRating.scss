.stars {
  display: flex;

  .star {
    width: 36px;
    height: 37px;
    margin-right: 5px;
    background-image: url("../../assets/images/star.png");
  }

  .rating {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 60px;
    height: 37px;
    border: 2px solid #09ADE3;
    border-radius: 5px;
    font-size: 24px;
  }
}
