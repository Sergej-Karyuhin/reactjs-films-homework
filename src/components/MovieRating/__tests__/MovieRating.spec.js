import React from 'react';
import renderer from 'react-test-renderer';
import MovieRating from '../MovieRating';

const rating = 2.5;

it('renders correctly', () => {
  const tree = renderer
    .create(<MovieRating movieRating={rating} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
