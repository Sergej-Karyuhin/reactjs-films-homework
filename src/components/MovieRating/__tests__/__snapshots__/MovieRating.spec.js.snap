// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="stars"
>
  <li
    className="star"
  />
  <li
    className="star"
  />
  <li
    className="star"
  />
  <div
    className="rating"
  >
    2.5
  </div>
</ul>
`;
